libdnf/module/ModulePackageContainer.cpp

```diff
@@ -1432,4 +1491,160 @@ ModulePackageContainer::Impl::ModulePersistor::getRemovedProfiles()
     return profiles;
 }
 
+void ModulePackageContainer::loadFailSafeData()
+{
+    auto persistor = pImpl->persistor->configs;
+    
+    std::map<std::string, std::pair<std::string, bool>> enabledStreams;
+    for (auto & nameConfig: persistor) {
+        if (nameConfig.second.second.state == ModuleState::ENABLED) {
+            auto & stream = nameConfig.second.second.stream;
+            if (!stream.empty()) {
+                enabledStreams.emplace(nameConfig.first, std::make_pair(stream, false));
+            }
+        }
+    }
+    for (auto & modulePair: pImpl->modules) {
+        auto module = modulePair.second.get();
+        auto it = enabledStreams.find(module->getName());
+        if (it != enabledStreams.end() && it->second.first == module->getStream()) {
+            it->second.second = true;
+        }
+    }
+    g_autofree gchar *path = g_build_filename(pImpl->installRoot.c_str(),
+                                              "/tmp/yaml", NULL);
+    auto fileNames = getYamlFilenames(path);
+    auto begin = fileNames.begin();
+    auto end = fileNames.end();
+    for (auto & pair: enabledStreams) {
+        if (!pair.second.second) {
+            // load from disk
+            std::ostringstream ss;
+            ss << pair.first << ":" << pair.second.first << ":";
+            bool loaded = false;
+            auto low = std::lower_bound(begin, end, ss.str(), stringStartWith);
+            for (auto iter = low; iter != end && stringStartWith(*iter, ss.str()); ++iter) {
```

Comment:
The `ss.str()` returns a string object with a copy of the current contents of the stream. Good idea is use the function once before loop.